client/components/domains/transfer-domain-step/index.jsx

```diff
@@ -630,29 +630,27 @@ class TransferDomainStep extends Component {
 		this.setState( { submittingWhois: true } );
 
 		return new Promise( ( resolve ) => {
-			checkInboundTransferStatus(
-				getFixedDomainSearch( this.state.searchQuery ),
-				( error, result ) => {
+			checkInboundTransferStatus( getFixedDomainSearch( this.state.searchQuery ) ).then(
+				( result ) => {
```

Comment:
...and here, `checkInboundTransferStatus` already returns a promise, so you don't need to create a new one with `new Promise`. Just transform the returned one.

Reply:
Yep, I'd thought/played with this too - thank you! I've pushed an updated version... I _think_ I'm returning the values properly, since the promise is already resolved when the function returns but I'm eager to hear your thoughts.

I _do_ now have a new error though. If the domain is still locked, clicking the 
I've unlocked my domain" button [throws an TypeError here](https://github.com/Automattic/wp-calypso/blob/f416951a071e78c3a2b0df691d553e9084336d18/client/components/domains/transfer-domain-step/transfer-domain-precheck.jsx#L88) because `this.props.refreshStatus( this.statusRefreshed )` returns as `undefined`.  I'm _very_ new to React, so I can't quite tell yet why that's happening, I'll need to investigate some more but welcome any advice.